omod/src/main/java/org/openmrs/web/rest/resource/TagResource.java

```
@@ -0,0 +1,266 @@
+	 */
+	@Override
+	public Tag getByUniqueId(String uuid) {
+		
+		try {
+			UUID uuid1 = UUID.fromString(uuid);
+			return Context.getService(TagService.class).getTagByUuid(uuid);
+		}
+		catch (IllegalArgumentException e) {
+			Tag tag = new Tag();
+			tag.setTag(uuid);
+			return tag;
+		}
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#save(Object)
+	 */
+	@Override
+	public Tag save(Tag tag) {
+		return getService().saveTag(tag);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#delete(Object,String, RequestContext)
+	 */
+	@Override
+	protected void delete(Tag tag, String reason, RequestContext context) {
+		throw new ResourceDoesNotSupportOperationException("delete not allowed on tag");
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#purge(Object, RequestContext)
+	 */
+	@Override
+	public void purge(Tag tag, RequestContext context) {
+		getService().purgeTag(tag);
+	}
+	
+	/**
+	 * Returns the DefaultRepresentation
+	 */
+	@RepHandler(DefaultRepresentation.class)
+	public SimpleObject asDef(Tag delegate) throws ConversionException, ClassNotFoundException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("tag");
+		if (StringUtils.isNotBlank(delegate.getObjectType())) {
+			description.addProperty("uuid");
+			description.addProperty("objectType");
+			description.addProperty("objectUuid");
+		}
+		List<Tag> tagList = Context.getService(TagService.class).getTags(delegate.getTag(), true);
+		Iterator<Tag> tagIterator = tagList.iterator();
+		while (tagIterator.hasNext()) {
+			Tag iterator = tagIterator.next();
+			description.addLink(getResourceName(iterator.getObjectType()),
+			    getUriOfObject(iterator.getObjectType(), iterator.getObjectUuid()));
+		}
+		description.addSelfLink();
+		description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		if (tagList.size() == 0) {
+			description = null;
```

A null description seems pointless to me, If there is no match in the database, I think it's better to fail loudly, this is how other resources work.